lib/phlex/sgml.rb

```diff
@@ -134,7 +139,12 @@ def __final_call__(buffer = +"", context: Phlex::Context.new, view_context: nil,
 				end
 			end
 
-			buffer << context.buffer unless parent
+			unless parent
+				if Phlex::SUPPORTS_FIBER_STORAGE
+					Fiber[:__phlex_component__] = original_fiber_storage
```

Comment:
Putting this back rather than setting it to `nil` is only necessary if you separately render a different component from scratch while rendering this component.